src/masternode.cpp

```diff
@@ -240,15 +239,34 @@ bool CMasternodeBroadcast::Create(const std::string& strService,
         return false;
     }
 
+    std::string strError;
+    if (strTxHash.empty() || strOutputIndex.empty()) {
+        strError = "Invalid masternode collateral hash or output index";
+        return error("%s: %s", __func__, strError);
+    }
+
+    const uint256 collateralHash = uint256S(strTxHash);
+    int collateralOutputIndex;
+    try {
+        collateralOutputIndex = std::stoi(strOutputIndex.c_str());
```

Comment:
nit: new introduction of a locale-dependent function

nit because this is both temporary code given the upcoming DMNs feature, and i've got a local branch to refactor this addition so as to eliminate this (and other) locale-dependent function usages revolving around the output index of MNs.

We currently don't have a linter for locale-dependent functions, so I'm not asking for a change here, just commenting to point it out for my own purposes. If anything, could add a `TODO:` comment if you so choose.

Reply:
Isn't a new addition, just moved the code from `wallet.cpp` to `masternode.cpp`. It's coming from `CWallet::GetMasternodeVinAndKeys`.

And yep, all what is inside `masternode.cpp` will be removed. The new `GetMasternodeVinAndKeys` will be used equally for legacy and deterministic Masternodes. This was a small decoupling coming from the DMN GUI branch.